### ISSUE TITLE
Dynamic resizing of component/iframe

### ISSUE THREAD

**Issue by Socvest** [lfoppiano/streamlit-pdf-viewer#79]
I have made a pull request regarding dynamically resizing the width of the component's iframe. Similar implementation was made using [this](https://github.com/victoryhb/streamlit-option-menu/blob/master/streamlit_option_menu/frontend/src/MyComponent.vue) component.

**Comment by stefanadelbert** [comment]
I would love to see this functionality included in the component.

At the moment I'm having to use [streamlit-dimensions](https://github.com/avsolatorio/streamlit-dimensions) to get the width of a container and then use that width to set the width of the pdf viewer. I have had to hack in some hysteresis (only change the width of the pdf viewer if the width of the container was changing by more than a certain amount to avoid continuous reruns). streamlit-dimensions also causes vertical space to be used on the screen.

**Comment by lfoppiano** [comment]
hi @stefanadelbert we're working on it, but it takes time since this is a side project.
There are failing tests that needs to be addressed before we can merge and release this feature. 😸

**Comment by lfoppiano** [comment]
I see everybody is quite interested in this feature, I've released a first preliminary version `0.0.21-dev0`, please try it and let me know if you have any issue.

**Comment by stefanadelbert** [comment]
I have just tried `0.0.21-rc0`. I am able to provide a percentage like "100%" to the width parameter. Unfortunately this doesn't do what I was hoping for... If I set the width to "50%" then the PDF is rendered in 50% of the width of the container, but if the window is resized, that width is not dynamically updated.

I was hoping for dynamic resizing which would make the PDF fit the width of the container as the width of that container changed, e.g. when a page is resized or the navigation bar on the left is closed/opened.

@lfoppiano Please let me know if you would like more information from me on this.

**Comment by Socvest** [comment]
@stefanadelbert right, you don't like the horizontal scroll, but rather wish to view the whole doc within lets say 50% of the width of the container as the window size changes? Is that I am getting from what you said?

**Comment by stefanadelbert** [comment]
Hi @Socvest @lfoppiano 

The effect that I'm trying to achieve is a side-by-side layout with PDF on the left and a form on the right. I'm using columns for this. I would like the PDF to "fill" the width of the left column, even when the width of the left column changes, e.g. when the window width changes or the side navigation is collapsed.

My idea if that I would put the pdf_viewer element in the left column with width "100%" and then when the window width changes, the column width changes and the pdf_view element width changes to fill the new column width.

Please let me know if this make sense or if you'd like more details. Very happy to discuss further!

:pray: :muscle:

**Comment by lfoppiano** [comment]
@stefanadelbert I think it make sense, a dynamic size in percentage should adapt to the window's size. In fact finding the right size of the window it was tricky but maybe there is a solution I did overlook. @t29mato any hint from your experience?

**Comment by lfoppiano** [comment]
Great! 🐯 🚀 

I'm releasing in version 0.0.21-rc5 (got some issues with the build)

**Comment by stefanadelbert** [comment]
I thought I had commented on this already. My apologies!

I have been running 0.0.21-rc5 in production since last week and it's been working great! 🙏 💪

**Comment by lfoppiano** [comment]
Implemented in version 0.0.21. 🎉
